modules/packages/LinearAlgebra.chpl

```diff
@@ -1042,6 +1042,190 @@ proc trace(A: [?D] ?eltType) {
   return trace;
 }
 
+proc _lu (in A: [?Adom] ?eltType) {
+  const n = Adom.shape(1);
+  const LUDom = {1..n, 1..n};
+
+  // TODO: Reduce memory usage
```

Comment:
Just to be clear, you are talking about wasted memory for storing triangular matrices as full matrices, right?

Reply:
Yes, we could possibly be wasting 2 times more memory with the current implementation